### PR TITLE
need to check lower bound on index in rw_range_sett::get_objects_byte_extract

### PR DESCRIPTION
**Pull request by kroening**
This addresses part of issue #2653

### CODE REVIEW

**Comment by peterschrammel** [comment]
This seems related to https://github.com/diffblue/cbmc/issues/2524.

**Comment by kroening** [comment]
Ok, let's see whether that can do a KNOWNBUG -> CORE.

File: src/analyses/goto_rw.cpp [hunk 1]
```diff
@@ -145,7 +145,8 @@ void rw_range_sett::get_objects_byte_extract(
   else
   {
     index*=8;
-    if(index>=pointer_offset_bits(be.op().type(), ns))
+    if(index<0 ||
+       index>=pointer_offset_bits(be.op().type(), ns))
```
Comment (tautschnig) [inline src/analyses/goto_rw.cpp]: I'm afraid this code has more flaws (yes, my fault): the offset must be computed relative to the root object, which may or may not be the same as `be.op()`. Consider the case of a byte_extract within a struct member (and the test that's now supposedly fixed is exactly of that nature).

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Comment by NlightNFotis** [comment]
Hi @kroening, is it possible that you address the issues that Michael raised?

If not, shall we have this closed please?

**Review by tautschnig — APPROVED** [review]

**Review by chris-ryder — APPROVED** [review]

**Pull request merged**